.github/workflows/node.js.yml

```diff
@@ -0,0 +1,59 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/*.js.yml
+      - src/js/**/*[tj]sx?
+      - tests/js/**/*.[tj]s
+      - package.json
+      - '*.js'
+  push:
```

Comment:
I still think that this list of branches mean that if I push a JS file change to a feature branch, it's not going to run the tests. 

Only when the PR is made, will they run, and at that time I thought it was all working fine, but it might not have been.

Reply:
If you push to a branch that's attached to a PR it will run by virtue of the `pull_request.synchronize` event. If you push to a branch that has not been PR'd, it won't run. Seems ok to me.